### DISCUSSION TITLE
Mashumaro instead of Pydantic

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#8820] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from dataclasses import dataclass
from typing import Optional

from fastapi import FastAPI
from mashumaro import DataClassJSONMixin

app = FastAPI()


@dataclass
class Item(DataClassJSONMixin):
    name: str
    price: float
    is_offer: Optional[bool] = None


@app.get("/")
def read_root():
    return {"Hello": "World"}


@app.get("/items/{item_id}")
def read_item(item_id: int, q: Optional[str] = None):
    return {"item_id": item_id, "q": q}


@app.put("/items/{item_id}")
def update_item(item_id: int, item: Item):
    return {"item_name": item.name, "item_id": item_id}
```

### Description

pydantic can be a bit on the slower side and the default functionality is not geared towards strict typing. `"1"` can become `1` in an `int()` field. I don't think this is good behavior for APIs. Also, mypy only catches pydantic issues with a plugin. Which does work well, but it is a bit unusual in my mind.

https://github.com/Fatal1ty/mashumaro looks like it provides the needed functionality, but with speed benefits and strictness.

I know it'd be a huge change, or maybe mashumaro could be supported in addition. In the past I've liked dacite, but it looks like mashumaro might be even better.

Just to note that as it is, dataclasses in FastAPI are very limited and buggy.

### Wanted Solution

Being able to use mashumaro classes in FastAPI.

### Wanted Code

```python
from dataclasses import dataclass
from typing import Optional

from fastapi import FastAPI
from mashumaro import DataClassJSONMixin

app = FastAPI()


@dataclass
class Item(DataClassJSONMixin):
    name: str
    price: float
    is_offer: Optional[bool] = None


@app.get("/")
def read_root():
    return {"Hello": "World"}


@app.get("/items/{item_id}")
def read_item(item_id: int, q: Optional[str] = None):
    return {"item_id": item_id, "q": q}


@app.put("/items/{item_id}")
def update_item(item_id: int, item: Item):
    return {"item_name": item.name, "item_id": item_id}
```

### Alternatives

dacite

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.8.10

### Additional Context

Thank you!

**Comment by Kludex** [comment, marked as answer]
Related to #3365

A similar request was asked some time ago about [Marshmallow](https://github.com/marshmallow-code/marshmallow) on #2685.

**Reply by stalkerg** [reply]
I don't think it's similar, because mashumaro faster and fresher. Optional serialization will be the best.

**Comment by adriangb** [comment]
If you look at how FastAPI processes Pydantic models, it's through the dependency injection system. To that end, I believe that decoupling the DI system from the rest of the framework would make it easier to define how to build an arbitrary class instance from request data.

**Comment by rominf** [comment]
There is also https://github.com/yukinarit/pyserde. It's not very popular and still in beta, but it's promising because it's even faster.

**Comment by adriangb** [comment]
> There is also https://github.com/yukinarit/pyserde. It's not very popular and still in beta, but it's promising because it's even faster.

That looks cool! One key feature that Pydantic has, that most of the other libraries are missing as far as I can tell, is JSON schema support and related types (e.g. `constr(regex=...)` correctly translates to the `pattern` JSON schema field), which is required in order for FastAPI to generate OpenAPI schemas.

**Comment by kevinheavey** [comment]
> One key feature that Pydantic has, that most of the other libraries are missing as far as I can tell, is JSON schema support and related types (e.g. `constr(regex=...)` correctly translates to the `pattern` JSON schema field), which is required in order for FastAPI to generate OpenAPI schemas.

Just chiming in to add that [apischema supports this.](https://wyfo.github.io/apischema/0.16/json_schema/)

**Comment by c00kiemon5ter** [comment]
apischema actually looks very good!
